### PR TITLE
pynut2: init at 2.1.2

### PR DESCRIPTION
**Pull request by LucaFulchir**
Added pynut2 and added home-assistant dependency to it

###### Motivation for this change

Home-assistant integration with Network Ups Tools was not working, this library was missing.  
Added it to nixpkg and added the dependency to home-assistant.

Tested and currently working on x86-64  
Note that python tests are disabled, due to broken tests (I think)  
I left that part commented out to avoid others wasting time trying to reintroduce tests  

to test in home-assistant, add a `nut: {}` section in the config (Again, working and reporting correct values)

This is my first contribution, feel free to comment on what to do better.
I added myself to the maintainers list as required from CONTRIBUTING.md

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6861,6 +6861,12 @@
     githubId = 13791;
     name = "Luke Gorrie";
   };
+  luker = {
+    email = "luker@fenrirproject.org";
+    github = "LucaFulchir";
+    githubId = 2486026;
+    name = "Luca Fulchir";
+  };
   lumi = {
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:6870]: Please make this into a separate commit:

```
maintainers: add <name>
```

File: pkgs/servers/home-assistant/component-packages.nix [hunk 2]
```diff
@@ -590,7 +590,7 @@
     "nuki" = ps: with ps; [ pynuki ];
     "numato" = ps: with ps; [ ]; # missing inputs: numato-gpio
     "number" = ps: with ps; [ ];
-    "nut" = ps: with ps; [ ]; # missing inputs: pynut2
+    "nut" = ps: with ps; [ pynut2 ];
```
Comment (jonringer) [inline pkgs/servers/home-assistant/component-packages.nix:593]: this should probably be a separate commit from the package addition

File: pkgs/development/python-modules/pynut2/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  #checkInputs = [
+  #  pytestCheckHook
+  #];
+
+  pythonImportsCheck = [ "pynut2.nut2" ];
+
+  meta = with lib; {
+    description = "API overhaul of PyNUT, a Python library to allow communication with NUT (Network UPS Tools) servers.";
+    homepage = "https://github.com/mezz64/python-nut2";
+    license = with licenses; [ gpl3 ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pynut2/default.nix:33]: License is [`GPLv3+`](https://github.com/mezz64/python-nut2/blob/master/setup.py#L18).

```suggestion
    license = with licenses; [ gpl3Plus ];
```

File: pkgs/development/python-modules/pynut2/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    rev = "${version}";
+    sha256 = "1lg7n1frndfgw73s0ssl1h7kc6zxm7fpiwlc6v6d60kxzaj1dphx";
+  };
+
+  propagatedBuildInputs = [
+    requests
+  ];
+
+  # tests are broken, they import 'nut2', not 'pynut2.nut2'
+  #checkInputs = [
```
Comment (fabaff) [inline pkgs/development/python-modules/pynut2/default.nix:24]: Something like this could help to get the tests running:

```bash
  postPatch = ''
    substituteInPlace tests/test_nutclient.py \
      --replace "from nut2 import PyNUTClient" "from pynut2.nut2 import PyNUTClient"
  '';
```

File: pkgs/development/python-modules/pynut2/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "pynut2";
+  version = "2.1.2";
+
+  src = fetchFromGitHub {
+    owner = "mezz64";
+    repo = "python-nut2";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pynut2/default.nix:15]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/pynut2/default.nix (hunk 4 above)
Comment (LucaFulchir) [reply pkgs/development/python-modules/pynut2/default.nix:24]: Thank you, but after a more in depth look I noticed it's a lost cause

The tests are for and old api of the lib. I tried playing with those a bit, but I am not a python programmer and it's well beyond a few substitutions

**Comment by LucaFulchir** [comment]
fixed it all, except tests, those are a lost cause.

The project also has a small PR for modernizing out of python2 style, but it's been sitting there for a few of months and the issue reporting is disabled on the project.  
I don't think much is going to happen there

**Comment by LucaFulchir** [comment]
ping to the home-assistant maintainers, does this need more work?
@Mic92 @mweinelt @globin @fabaff

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 145204` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.pynut2</li>
<li>python39Packages.pynut2</li>
</ul>
</details>

File: pkgs/development/python-modules/pynut2/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+  # tests are completely broken, wrong imports and old api
+  #checkInputs = [
+  #  pytestCheckHook
+  #];
+
+  pythonImportsCheck = [ "pynut2.nut2" ];
+
+  meta = with lib; {
+    description = "API overhaul of PyNUT, a Python library to allow communication with NUT (Network UPS Tools) servers.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pynut2/default.nix:31]: No period: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes
